packages/core/src/state.js

```diff
@@ -3,9 +3,12 @@ import camelcase from 'camelcase'
 
 function getComponentName(state) {
   if (!state.filePath) return 'SvgComponent'
-  const pascalCaseFileName = camelcase(path.parse(state.filePath).name, {
-    pascalCase: true,
-  })
+  const pascalCaseFileName = camelcase(
+    path.parse(state.filePath).name.replace(/[^a-zA-Z0-9_-]/g, ''),
```

Comment:
Can you please extract the RegExp in a constant outside the function (it is faster).

Reply:
👍